httpmw/ip_cognito_auth.go

```diff
@@ -193,24 +192,41 @@ func IpCognitoAuth(ipRange string, svc cognitoidentityprovideriface.CognitoIdent
 			return
 		}
 
-		model, ok := ch.Get(token)
+		data, err := cache.Get(c, token).Bytes()
 
-		if v, ok := model.(*CognitoUser); ok {
-			user = v
+		if err != nil && err != redis.Nil {
+			httperr.InternalServerError(c, err.Error())
+			c.Abort()
+			return
+		}
+
+		if err == nil {
+			if err := json.Unmarshal(data, user); err != nil {
+				httperr.InternalServerError(c, err.Error())
+				c.Abort()
+				return
+			}
 		}
 
-		if !ok {
+		if err == redis.Nil {
 			res, err := svc.GetUser(&cognitoidentityprovider.GetUserInput{AccessToken: &token})
 
 			if err != nil {
-				log.Println(err)
-				httperr.Unauthorized(c)
+				httperr.Unauthorized(c, err.Error())
 				c.Abort()
 				return
 			}
 
 			user.SetUsername(*res.Username)
-			ch.SetDefault(token, user)
+			data, err := json.Marshal(user)
+
+			if err != nil {
+				httperr.InternalServerError(c, err.Error())
+				c.Abort()
+				return
+			}
+
+			cache.Set(c, token, data, expire)
```

Comment:
We should probably make cache key something like `fmt.Sprintf("token:%s", token)` so we have at least a little bit of namespaceng for our keys.